src/main/scala/ai/privado/languageEngine/ruby/processor/RubyProcessor.scala

```diff
@@ -62,6 +64,12 @@ object RubyProcessor {
         logger.info("Enhancing Ruby graph")
         logger.debug("Running custom passes")
 
+        if (config.skipDownloadDependencies) {
+          val packageTable = new PackageTable()
+          println(s"${Calendar.getInstance().getTime} - Downloading dependencies and parsing ...")
+          ExternalDependenciesResolver.downloadDependencies(cpg, sourceRepoLocation, packageTable)
```

Comment:
Instead of passing packageTable, let's return packageTable from downloadDependencies function, so that function `downloadDependencies` is a pure function

Reply:
okay sure.